doc/theoretical_description_conformity_scores.rst

```diff
@@ -0,0 +1,113 @@
+methods to compute conformity scores for regression.
+We give here a brief theoretical description of the scores included in the module.
+Note that it is possible for the user to create any conformal scores that are not 
+already included in MAPIE by inheriting this class.
+
+Before describing the methods, let's briefly present the mathematical setting.
+With conformal predictions we want to transform an heuristic notion of uncertainty
+from a model into a rigorous one, and the first step to do it is to choose a conformal score.
+The only requirement for the score function :math:`s(X, Y) \in \mathbb{R}` is
+that larger scores should encode worse agreement between :math:`X` and :math:`Y`. [1]
+
+There are two types of scores : the symmetric and asymmetric ones.
+The symmetric property defines the way of computing the quantile of the conformity
+scores when calculating the interval's bounds. If a score is symmetrical two
+quantiles will be computed : one on the right side of the distribution
+and the other on the left side.
+
+1. The absolute residual score
+==============================
+
+The absolute residual score (:class:`mapie.conformity_scores.AbsoluteResidualScore`)
+is the simplest and most commonly used conformal score, it translates the error
+of the model : in regression it is called the residual.
+
+.. math:: |Y-\hat{\mu}(X)|
+
+The intervals of prediction's bounds are then computed from the following formula :
+
+.. math:: [\hat{\mu}(X) - q(s), \hat{\mu}(X) + q(s)]
+
+Where :math:`q(s)` is the :math:`(1-\alpha)` quantile of the conformity scores.
+(see :doc:`theoretical_description_regression` for more details).
+
+With this score the intervals of predictions will be constant over the whole dataset.
+This score is by default symmetric (*see above for definition*).
+
+2. The gamma score
+==================
+
+The gamma score (:class:`mapie.conformity_scores.GammaConformityScore`) adds a
+notion of adaptivity with the normalization of the residuals by the predictions.
+
+.. math:: \frac{|Y-\hat{\mu}(X)|}{\hat{\mu}(X)}
+
+It computes adaptive intervals : intervals of different size on each example, with
+the following formula  :
+
+.. math:: [\hat{\mu}(X) * (1 - q(s)), \hat{\mu}(X) * (1 + q(s))]
+
+Where :math:`q(s)` is the :math:`(1-\alpha)` quantile of the conformity scores.
+(see :doc:`theoretical_description_regression` for more details).
+
+This score is by default asymmetric (*see definition above*).
+
+Compared to the absolute residual score, it allows to see regions with smaller intervals
+than others which are interpreted as regions with more certainty than others.
+It is important to note that, this conformity score is inversely proportional to the
+order of magnitude of the predictions. Therefore, the uncertainty is proportional to
+the order of magitude of the predictions, implying that this core should be used
+in use cases where we want greater uncertainty when the prediction is high.
```

Comment:
Are these insights based on a paper? Or can we link to some example we have?

Reply:
This is the traduction of the score formula
I can link this paper : "Cordier, T., Blot, V., Lacombe, L., Morzadec, T., Capitaine, A. &amp; Brunel, N.. (2023). Flexible and Systematic Uncertainty Estimation with Conformal Prediction via the MAPIE library"